### ISSUE TITLE
Issue with version cucumber@1.2.1 commit - StackTrace getFileName of undefined

### ISSUE THREAD

**Issue by gkushang** [cucumber/cucumber-js#610]
We are seeing below issue with the recent release `1.2.1` and possibly with this commit - https://github.com/cucumber/cucumber-js/commit/1d02c6af66763189ef8ffc18880245ce39a56172. 

Below is stackTrace,

```
TypeError: Cannot call method 'getFileName' of undefined
    at isFrameInCucumber (.../node_modules/cucumber/lib/cucumber/runtime/stack_trace_filter.js:9:24)
    at Array.0 (.../node_modules/cucumber/lib/cucumber/runtime/stack_trace_filter.js:15:9)
    at TraceModifier._modify (...node_modules/stack-chain/stack-chain.js:44:32)
    at Function.prepareStackTrace (.../node_modules/cucumber/node_modules/stack-chain/stack-chain.js:119:25)
    at promise.Promise.resolve_ (eval at <anonymous> (.../node_modules/selenium-webdriver/lib/goog/base.js:1094:19), <anonymous>:440:36)
    at eval (eval at <anonymous> (.../node_modules/selenium-webdriver/lib/goog/base.js:1094:19), <anonymous>:371:12)
    at reject (eval at <anonymous> (.../node_modules/selenium-webdriver/lib/goog/base.js:1094:19), <anonymous>:709:5)
    at promise.ControlFlow.runInFrame_ (eval at <anonymous> (.../node_modules/selenium-webdriver/lib/goog/base.js:1094:19), <anonymous>:1917:5)
    at goog.defineClass.notify (eval at <anonymous> (.../node_modules/selenium-webdriver/lib/goog/base.js:1094:19), <anonymous>:2448:25)
   ...

```
Previous releases works fine.

**Comment by gkushang** [comment]
Below is our observations:

```
1. `this.Then` step has chai-as-promised assertions

2. Assertion fails. e.g. 

var chai = require('chai');
var chaiAsPromise = require('chai-as-promised');
var should = chai.should();

chai.use(chaiAsPromise);

//THEN STEP
    seleniumPage.getErrorMessage().should.eventually.to
      .equal('somethingNotExpectedString', 'failure message')
      .notify(function(err) {
        if (err) {
          callback(err);
        }
      });

3. Above error is reproducible when next step (after `this.Then`) executes, it can be either @After or @anotherGWT

```
 
This seems to be breaking change in recent version

HTH

Thanks,
Kushang

**Comment by charlierudolph** [comment]
I believe this was fixed in #609. I'll try to release 1.2.2 today

**Comment by gkushang** [comment]
Sounds great, thanks! Will reopen the issue if it still exists in upcoming version. 

Kushang
